extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/registry/json/JsonMeterRegistry.java

```diff
@@ -0,0 +1,118 @@
+
+    public JsonMeterRegistry(Clock clock) {
+        super(clock);
+        this.jsonExporter = new JsonExporter();
+    }
+
+    @Override
+    protected <T> Gauge newGauge(Meter.Id id, T obj, ToDoubleFunction<T> valueFunction) {
+        return new DefaultGauge<>(id, obj, valueFunction);
+    }
+
+    @Override
+    protected Counter newCounter(Meter.Id id) {
+        return new CumulativeCounter(id);
+    }
+
+    @Override
+    protected Timer newTimer(Meter.Id id, DistributionStatisticConfig distributionStatisticConfig,
+            PauseDetector pauseDetector) {
+        // turn off percentiles and histograms because we don't need them for the JSON export
+        distributionStatisticConfig = DistributionStatisticConfig.builder()
+                .percentilesHistogram(false)
+                .percentiles(new double[0])
+                .build()
+                .merge(distributionStatisticConfig);
+        return new JsonTimer(id, clock, distributionStatisticConfig, pauseDetector, getBaseTimeUnit());
+    }
+
+    @Override
+    protected DistributionSummary newDistributionSummary(Meter.Id id, DistributionStatisticConfig distributionStatisticConfig,
+            double scale) {
+        distributionStatisticConfig = DistributionStatisticConfig.builder()
+                .percentilePrecision(3)
+                .percentilesHistogram(false)
+                .minimumExpectedValue(Double.MIN_VALUE)
+                .maximumExpectedValue(Double.MAX_VALUE)
+                .bufferLength(1024)
+                .expiry(Duration.ofDays(3))
+                .build()
+                .merge(distributionStatisticConfig);
+        return new CumulativeDistributionSummary(id, clock, distributionStatisticConfig, scale, false);
+    }
+
+    @Override
+    protected Meter newMeter(Meter.Id id, Meter.Type type, Iterable<Measurement> measurements) {
+        return new DefaultMeter(id, type, measurements);
+    }
+
+    @Override
+    protected <T> FunctionTimer newFunctionTimer(Meter.Id id, T obj, ToLongFunction<T> countFunction,
+            ToDoubleFunction<T> totalTimeFunction, TimeUnit totalTimeFunctionUnit) {
+        return new CumulativeFunctionTimer<>(id, obj, countFunction, totalTimeFunction,
+                totalTimeFunctionUnit, getBaseTimeUnit());
+    }
+
+    @Override
+    protected <T> FunctionCounter newFunctionCounter(Meter.Id id, T obj, ToDoubleFunction<T> countFunction) {
+        return new CumulativeFunctionCounter<>(id, obj, countFunction);
+    }
+
+    @Override
+    protected LongTaskTimer newLongTaskTimer(Meter.Id id, DistributionStatisticConfig distributionStatisticConfig) {
+        return new NoopLongTaskTimer(id);
+    }
+
+    @Override
+    protected TimeUnit getBaseTimeUnit() {
+        return TimeUnit.MILLISECONDS;
+    }
+
+    @Override
+    protected DistributionStatisticConfig defaultHistogramConfig() {
+        return DistributionStatisticConfig.builder()
+                .percentiles(0.5, 0.75, 0.95, 0.98, 0.99)
```

Comment:
p50 getSnapshot().getMedian()
p75 getSnapshot().get75thPercentile()
p95 getSnapshot().get95thPercentile()
p98 getSnapshot().get98thPercentile()
p99 getSnapshot().get99thPercentile()
p999 getSnapshot().get999thPercentile()

?

Reply:
One thing we could consider is not publishing the percentiles by default (but publish the other calculations), and rely on the usual MeterFilter behavior for users to turn those on.. the memory implications for using histograms isn't trivial.

Reply:
I added the missing 99.9%.
Whether to publish percentiles by default, I'd say yes, since we are trying to emulate the MP Metrics exporter, which does that always :rabbit:

Reply:
This one is resolved, too. ;)